src/lex.cpp

```diff
@@ -256,7 +256,10 @@ void lexer::parse_current_token() {
     break;
 
   case '<':
-    if (this->input_[1] == '=') {
+    if (this->input_[1] == '!') {
+      this->skip_html_line_comment();
```

Comment:
Almost-blocking: This would break code like `x<!y` right? (I think `x<!y` should be parsed as `x` `<` `!` `y`, not as `x` followed by an HTML line comment.) Maybe `x<!-y` too. (Perhaps these cases deserves some tests!)

Reply:
Yes, this would break.
I tried fixing it by adding more checks for `-` and `-`, is that ok?
Also added 2 tests.